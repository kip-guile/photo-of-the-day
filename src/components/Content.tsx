import React from 'react'
import { ContentContainer } from '../styles/styles'
import { Collapse } from 'antd'

const { Panel } = Collapse

// Describe Content prop structure
interface ContentProps {
  explanation: string
}

const Content = ({ explanation }: ContentProps) => {
  return (
    <ContentContainer>
      <Collapse defaultActiveKey={['1']}>
        <Panel header='Photo Description' key='1'>
          <p>{explanation}</p>
        </Panel>
      </Collapse>
    </ContentContainer>
  )
}

export default Content
