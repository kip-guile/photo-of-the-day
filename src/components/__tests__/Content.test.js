import React from 'react'
import * as rtl from '@testing-library/react'
import Content from '../Content'
import '@testing-library/jest-dom/extend-expect'
import { create, act } from 'react-test-renderer'

let tools

beforeEach(() => {
  rtl.cleanup()
  tools = rtl.render(<Content explanation='string' />)
})

//testing the content UI component
it('renders without crashing', () => {
  let res
  act(() => {
    res = create(<Content explanation='string' />)
  })
  expect(res.toJSON()).toMatchSnapshot()
})

it('displays content', () => {
  const content = tools.queryByText(/string/i)
  expect(content).toBeVisible()
})
