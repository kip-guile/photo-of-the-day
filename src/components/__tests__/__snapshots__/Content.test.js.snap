// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="sc-iCoHVE eJNUsI"
>
  <div
    className="ant-collapse ant-collapse-icon-position-left"
    role={null}
  >
    <div
      className="ant-collapse-item ant-collapse-item-active"
    >
      <div
        aria-expanded="true"
        className="ant-collapse-header"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          aria-label="right"
          className="anticon anticon-right ant-collapse-arrow"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            style={
              Object {
                "msTransform": "rotate(90deg)",
                "transform": "rotate(90deg)",
              }
            }
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
            />
          </svg>
        </span>
        Photo Description
      </div>
      <div
        className="ant-collapse-content ant-collapse-content-active"
        role={null}
      >
        <div
          className="ant-collapse-content-box"
        >
          <p>
            string
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
